src/bpftrace.cpp

```diff
@@ -282,8 +282,8 @@ int BPFtrace::add_probe(ast::Probe &p)
         {
           Probe probe_copy = probe;
           probe_copy.usdt_location_idx = i;
-          probe_copy.index = attach_point->index(func + "_loc" +
-                                                 std::to_string(i));
+          probe_copy.index = attach_point->index() > 0 ? attach_point->index()
+                                                       : p.index();
```

Comment:
Shouldn't this assign different indices to different locations?

Reply:
The sections generated for the USDT probe clone are distinguished by name, see this passage in `CodegenLLVM::visit(Probe &probe)`:
```cpp
          current_usdt_location_index_ = 0;
          for (int i = 0; i < attach_point->usdt.num_locations; ++i)
          {
            reset_ids();

            std::string full_func_id = match + "_loc" + std::to_string(i);
            generateProbe(probe, full_func_id, probefull_, func_type, true, i);
            current_usdt_location_index_++;
          }
```
Therefore, they don't need different indices (also they can't have different indices, otherwise they wouldn't be properly matched to the section, which has the attach point index common to the clone and the original program).  Actually, the previous implementation assignes indices unique in the context of a probe name (it was made to fix an issue with multiple identical probes, see #100), hence its index is even less unique globally.

Reply:
Thanks, that makes sense.